plugins/org.orbisgis.groovyeditor/src/org/orbisgis/ui/editors/groovy/GroovyCompletionProcessor.java

```diff
@@ -0,0 +1,217 @@
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License
+ * for more details http://www.gnu.org/licenses.
+ *
+ *
+ *For more information, please consult: http://www.orbisgis.org
+ *or contact directly: info_at_orbisgis.org
+ */
+package org.orbisgis.ui.editors.groovy;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import org.eclipse.jface.text.IDocument;
+import org.eclipse.jface.text.ITextViewer;
+import org.eclipse.jface.text.contentassist.CompletionProposal;
+import org.eclipse.jface.text.contentassist.ContextInformation;
+import org.eclipse.jface.text.contentassist.ICompletionProposal;
+import org.eclipse.jface.text.contentassist.IContentAssistProcessor;
+import org.eclipse.jface.text.contentassist.IContextInformation;
+import org.eclipse.jface.text.contentassist.IContextInformationValidator;
+import org.eclipse.lsp4j.CompletionItem;
+import org.eclipse.lsp4j.CompletionList;
+import org.eclipse.lsp4j.CompletionParams;
+import org.eclipse.lsp4j.DidOpenTextDocumentParams;
+import org.eclipse.lsp4j.MessageActionItem;
+import org.eclipse.lsp4j.MessageParams;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.PublishDiagnosticsParams;
+import org.eclipse.lsp4j.ShowMessageRequestParams;
+import org.eclipse.lsp4j.TextDocumentIdentifier;
+import org.eclipse.lsp4j.TextDocumentItem;
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
+import org.eclipse.lsp4j.services.LanguageClient;
+
+import net.prominic.groovyls.GroovyServices;
+import net.prominic.groovyls.config.CompilationUnitFactory;
+
+/**
+ * Methods to create the list of the suggested autocompletion words.
+ *
+ * @author Adrien Bessy, CNRS
+ */
+public class GroovyCompletionProcessor implements IContentAssistProcessor {
+
+	/**
+	* Build the list of the completion proposals provided from the Groovy Langage Server project according to the word type.
+    * @param viewer the editor view
+    * @param offset the cursor position in the document
+    * @return the list of the completion proposals
+    */
+	@Override
+	public ICompletionProposal[] computeCompletionProposals(ITextViewer viewer, int offset) {
+		IDocument document = viewer.getDocument();
+		int currOffset = offset-1;
+		String currWord = "";
+		if (currOffset >= 0) {
+		    try {
+				char currChar;
+				/*
+				 * Find the start position of the word where we trigger the completion.
+				 * We move from offset to offset until we meet a space or \n character.
+				 * Each character is added at the start of the current word.
+				 */
+				while (currOffset >= 0 && !Character.isWhitespace(currChar = document.getChar(currOffset))) {
+				    currWord = currChar + currWord;
+				    System.out.println("currWord : " + currWord);
+				    currOffset--;
+				}
+				
+				String LANGUAGE_GROOVY = "groovy";
```

Comment:
For magic constants, you better declare it as `private static final` variable in your class instead of in your method body.